src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -1117,36 +1117,36 @@ void WorldSession::HandleInspectHonorStatsOpcode(WorldPacket& recv_data)
     SendPacket(&data);
 }
 
-void WorldSession::HandleWorldTeleportOpcode(WorldPacket& recv_data)
+void WorldSession::HandleWorldTeleport(WorldPacket& message)
 {
-    uint32 time;
-    uint32 mapid;
-    float PositionX;
-    float PositionY;
-    float PositionZ;
-    float Orientation;
+    Position    position;
+    uint32      continentID;
+    uint32      requestTimeMs;
+    uint64      player;
 
-    recv_data >> time;                                      // time in m.sec.
-    recv_data >> mapid;
-    recv_data >> PositionX;
-    recv_data >> PositionY;
-    recv_data >> PositionZ;
-    recv_data >> Orientation;                               // o (3.141593 = 180 degrees)
+    if (!AccountMgr::IsGMAccount(GetSecurity()))
+    {
+        SendNotification(LANG_YOU_NOT_HAVE_PERMISSION);
+        return;
+    }
 
-    LOG_DEBUG("network", "WORLD: Received CMSG_WORLD_TELEPORT");
+    message >> requestTimeMs;
+    message >> continentID;
+    message >> player;
+    message >> position.m_positionX;
+    message >> position.m_positionY;
+    message >> position.m_positionZ;
+    message >> position.m_orientation;
 
-    if (GetPlayer()->IsInFlight())
+    Player* playerPtr = player ? ObjectAccessor::FindPlayer((ObjectGuid)player) : GetPlayer();
+
+    if (!playerPtr)
     {
-        LOG_DEBUG("network", "Player '{}' ({}) in flight, ignore worldport command.", GetPlayer()->GetName(), GetPlayer()->GetGUID().ToString());
+        SendNotification("Player not found");
```

Comment:
```suggestion
        SendNotification("WorldTeleport: Player not found");
```

Reply:
I am sending the Player not found notification as that is what the server would normally send on Blizzard realms.

Reply:
Is this a debug server-only type string or is this getting sent to players? Because if it's getting sent to players, I feel like there should be some localize-able way to send it.

Reply:
Unfortunately it seems like those kinds of things were not localized by Blizzard.